rotkehlchen/db/dbhandler.py

```diff
@@ -949,6 +954,277 @@ def delete_adex_events_data(self) -> None:
+            query += 'AND '.join(filters)
+
+        query, bindings = form_query_to_filter_timestamps(
+            query=query,
+            timestamp_attribute='timestamp',
+            from_ts=from_timestamp,
+            to_ts=to_timestamp,
+        )
+        results = cursor.execute(query, bindings)
+
+        events = []
+        for event_tuple in results:
+            try:
+                event = BalancerEvent.deserialize_from_db(event_tuple)
+            except DeserializationError as e:
+                self.msg_aggregator.add_error(
+                    f'Error deserializing Balancer event from the DB. Skipping event. '
+                    f'Error was: {str(e)}',
+                )
+                continue
+            events.append(event)
+
+        return events
+
+    def get_balancer_pools(
+            self,
+            addresses: Optional[List[ChecksumEthAddress]] = None,
+    ) -> List[BalancerEventPool]:
+        """Returns a list of Balancer pools optionally filtered by addresses"""
+        cursor = self.conn.cursor()
+        query = (
+            'SELECT '
+            'address, '
+            'tokens_number, '
+            'is_token0_unknown, '
+            'token0_address, '
+            'token0_symbol, '
+            'token0_name, '
+            'token0_decimals, '
+            'token0_weight, '
+            'is_token1_unknown, '
+            'token1_address, '
+            'token1_symbol, '
+            'token1_name, '
+            'token1_decimals, '
+            'token1_weight, '
+            'is_token2_unknown, '
+            'token2_address, '
+            'token2_symbol, '
+            'token2_name, '
+            'token2_decimals, '
+            'token2_weight, '
+            'is_token3_unknown, '
+            'token3_address, '
+            'token3_symbol, '
+            'token3_name, '
+            'token3_decimals, '
+            'token3_weight, '
+            'is_token4_unknown, '
+            'token4_address, '
+            'token4_symbol, '
+            'token4_name, '
+            'token4_decimals, '
+            'token4_weight, '
+            'is_token5_unknown, '
+            'token5_address, '
+            'token5_symbol, '
+            'token5_name, '
+            'token5_decimals, '
+            'token5_weight, '
+            'is_token6_unknown, '
+            'token6_address, '
+            'token6_symbol, '
+            'token6_name, '
+            'token6_decimals, '
+            'token6_weight, '
+            'is_token7_unknown, '
+            'token7_address, '
+            'token7_symbol, '
+            'token7_name, '
+            'token7_decimals, '
+            'token7_weight '
+            'FROM balancer_pools '
+        )
+        bindings = []
+        if addresses is not None and len(addresses) != 0:
+            questionmarks = '?' * len(addresses)
+            query += f'WHERE address IN ({",".join(questionmarks)});'
+            bindings.extend(addresses)
```

Comment:
Would probably make sense to have this behavior and some other places around tested.

Very little, and not too much since this will all be refactored and tests will also be deleted.

If time does not allow since we want the release **ASAP** then at least make sure it's manually tested.

Reply:
It has been manually tested with no addresses, addresses as empty list and addresses as non-empty list.